### ISSUE TITLE
SIARDDK: Scientific notation is not allowed, when exporting decimal numbers to xml file.

### ISSUE THREAD

**Issue by ThomasKristensen** [keeps/db-preservation-toolkit#121]
I'm using https://www.w3.org/TR/xmlschema-2/#decimal as a reference (it that the correct reference?).

"Bekendtgørelse om arkiveringsversioner" (Bekendtgoerelse-nr-1007-af-20-august-2010.pdf), p. 10 states that all decimal types must be mapped to XML-datatype 'decimal'. 

When exporting from PostgreSQL 9.5, db-type 'real' can be exported using scientific notation. See column c5 in the example below: 

`<row>
    <c1>RUS</c1>
    <c2>Russian\u0020Federation</c2>
    <c3>Europe</c3>
    <c4>Eastern\u0020Europe</c4>
    <c5>1.70754E7</c5
    <c6>1991</c6>
    <c7>146934000</c7>
    <c8>67.2</c8>
    <c9>276608.00</c9>
    <c10>442989.00</c10>
    <c11>Rossija</c11>
    <c12>Federal\u0020Republic</c12>
    <c13>Vladimir\u0020Putin</c13>
    <c14>3580</c14>
    <c15>RU</c15>
  </row>
`

We probably need to parse the decimal number and setup a DecimalFormat to make sure scientific notation is not used.

https://docs.oracle.com/javase/7/docs/api/java/text/DecimalFormat.html

https://github.com/magenta-aps/db-preservation-toolkit/blob/master/dbptk-core/src/main/java/com/databasepreservation/modules/siard/out/content/SIARDDKContentExportStrategy.java#L258

**Comment by chalkos** [comment]
@andreaskring Could you look into this?

**Comment by andreaskring** [comment]
Yes, I will fix this

2016-02-04 17:27 GMT+01:00 Bruno Ferreira <notifications@github.com>:

> @andreaskring <https://github.com/andreaskring> Could you look into this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/keeps/db-preservation-toolkit/issues/121#issuecomment-179930640>
> .
>



-- 
Andreas Kring
Magenta Aps
Incuba, Åbogade 15
DK-8200 Aarhus N

tel +45 3336 9699
dir +45 8177 1698

http://www.magenta.dk
andreas@magenta.dk

**Comment by ThomasKristensen** [comment]
@andreaskring I forgot to mention that I used the (Xml-) DatatypeConverter linked to below for parsing the values of xsd type 'hexBinary' and that this class also contains print-methods for decimal numbers etc.

see:
https://docs.oracle.com/javase/8/docs/api/javax/xml/bind/DatatypeConverter.html

**Comment by andreaskring** [comment]
@ThomasKristensen Ok - thank you for the link. I will have a look at this

**Comment by andreaskring** [comment]
@chalkos This problem is a general problem also concerning SIARD 1.0 and SIARD 2.0, since they also map decimal numbers to the XML-datatype "decimal", i.e. when exporting reals in scientific notation from a PostgreSQL to all of the SIARD formats, we will have this problem.

In order to fix this, we have to parse the decimal numbers here:
https://github.com/keeps/db-preservation-toolkit/blob/master/dbptk-core/src/main/java/com/databasepreservation/modules/postgreSql/in/PostgreSQLJDBCImportModule.java#L451

Would you still like me to look at this, or do you prefer to have a look at it your self, since this is in your part of the code?

**Comment by chalkos** [comment]
The problem is in the PostgreSQL import module. I'll handle this issue then.

**Comment by chalkos** [comment]
@andreaskring do you have the database to test this if I fix it?

**Comment by andreaskring** [comment]
@chalkos No, sorry. I do not have a DB for testing this. But maybe it can be tested by making a database in PostgreSQL containing a single table with a singe row containing a very large or small number of the type 'real'?

**Comment by chalkos** [comment]
Yes, that was my second option ;)
I'll probably do this in a test driven dev way. thanks!
